presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
         assertEquals(getCacheStats().getRemoteReads(), intermittentRemoteReads);
     }
 
+    @Test
+    public void testLargeFile()
+            throws Exception
+    {
+        byte[] randomData = new byte[(int) LARGE_FILE_SIZE.toBytes()];
+        new Random().nextBytes(randomData);
+
+        Path file = getStoragePath("large_file");
+        try (FSDataOutputStream outputStream = nonCachingFileSystem.create(file)) {
+            outputStream.write(randomData);
+        }
+
+        long beforeRemoteReads = getCacheStats().getRemoteReads();
+        long beforeCachedReads = getCacheStats().getCachedReads();
+
+        assertTrue(Arrays.equals(randomData, readFileBytes(cachingFileSystem.open(file))));
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
```

Comment:
Do we always have to wait 1s? Would polling for stats in shorter intervals (like 100ms) make sense?

Reply:
With 100ms I meant polling in a loop. Not wait single interval.